game-app/game-core/src/test/java/games/strategy/ui/UtilTest.java

```diff
@@ -2,12 +2,58 @@
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 import java.awt.Polygon;
+import java.lang.reflect.InvocationTargetException;
+import javax.swing.SwingUtilities;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 
 final class UtilTest {
+  Throwable inThreadException;
+
+  void runInEvenDispatchThreadAndRethrow(final Runnable run) throws Throwable {
+    inThreadException = null;
+    final Thread threadNotEventDispatchThread =
+        new Thread(
+            () -> {
+              try {
+                SwingUtilities.invokeAndWait(run);
+              } catch (final InterruptedException e) {
+                inThreadException = e;
+              } catch (final InvocationTargetException e) {
+                inThreadException = e.getTargetException();
+              }
+            });
+
+    threadNotEventDispatchThread.start();
+    try {
+      threadNotEventDispatchThread.join();
+    } catch (final InterruptedException e) {
+      inThreadException = e;
+    }
+    if (inThreadException != null) {
+      throw inThreadException;
+    }
+  }
```

Comment:
Ok this definitely works, but I'm curious why you explicitly chose this approach?
Is there any reason
```
void exampleTest() {
  assertDoesNotThrow(() -> SwingUtilities.invokeAndWait(Util::ensureOnEventDispatchThread));
}
```
does not work as expected?

If there is a specific reason, I'd recommend using a local `AtomicReference<Throwable>` variable instead of the `inThreadException` field

Reply:
If you are asking for the reason of using an instance variable inThreadException and separate method that is executing the thread and rethrowing any exception, then I can answer you that it seemed more general so we can specifically check for the expected exception.
Thanks for the hint with the AtomicReference I will see to incorporate this asap.

Reply:
Done.

Reply:
I was mainly wondering why you have to spawn a new thread in the first place, to my knowledge you can just call `invokeAndWait` directly?